kubernetes/istio_details_service.go

```diff
@@ -837,50 +837,67 @@ func ValidateVirtualServiceGateways(spec map[string]interface{}, gatewayNames ma
 	return true, -1
 }
 
-func MeshPolicyHasMTLSEnabled(meshPolicy IstioObject) bool {
-
+func PolicyHasMTLSEnabled(policy IstioObject) (bool, string) {
 	// It is mandatory to have default as a name
-	if meshMeta := meshPolicy.GetObjectMeta(); meshMeta.Name != "default" {
-		return false
+	if policyMeta := policy.GetObjectMeta(); policyMeta.Name != "default" {
+		return false, ""
 	}
 
 	// It is no globally enabled when has targets
-	targets, targetPresent := meshPolicy.GetSpec()["targets"]
+	targets, targetPresent := policy.GetSpec()["targets"]
 	specificTarget := targetPresent && len(targets.([]interface{})) > 0
 	if specificTarget {
-		return false
+		return false, ""
 	}
 
 	// It is globally enabled when a peer has mtls enabled
-	peers, peersPresent := meshPolicy.GetSpec()["peers"]
+	peers, peersPresent := policy.GetSpec()["peers"]
 	if !peersPresent {
-		return false
+		return false, ""
 	}
 
 	for _, peer := range peers.([]interface{}) {
 		peerMap := peer.(map[string]interface{})
 		if mtls, present := peerMap["mtls"]; present {
 			if mtlsMap, ok := mtls.(map[string]interface{}); ok {
 				// mTLS enabled in case there is an empty map or mode is STRICT
-				if mode, found := mtlsMap["mode"]; !found || mode == "STRICT" {
-					return true
+				if mode, found := mtlsMap["mode"]; found {
+					if mode == "STRICT" {
+						return true, "ENABLED"
+					} else {
+						return false, "DISABLED"
+					}
+				} else {
+					// STRICT mode when mtls object is empty
+					return true, "ENABLED"
 				}
 			} else {
-				// mTLS enabled in case mtls object is empty
-				return true
+				// STRICT mode when mtls object is empty
+				return true, "ENABLED"
 			}
 		}
 	}
 
-	return false
+	return false, ""
 }
 
-func DestinationRuleHasMeshWideMTLSEnabled(destinationRule IstioObject) bool {
+func DestinationRuleHasMeshWideMTLSEnabled(destinationRule IstioObject) (bool, string) {
 	// Following the suggested procedure to enable mesh-wide mTLS, host might be '*.local':
 	// https://istio.io/docs/tasks/security/authn-policy/#globally-enabling-istio-mutual-tls
+	return DestinationRuleHasMTLSEnabledForHost("*.local", destinationRule)
+}
+
+func DestinationRuleHasNamespaceWideMTLSEnabled(namespace string, destinationRule IstioObject) (bool, string) {
+	// Following the suggested procedure to enable namespace-wide mTLS, host might be '*.namespace.svc.cluster.local'
+	// https://istio.io/docs/tasks/security/authn-policy/#namespace-wide-policy
+	nsHost := fmt.Sprintf("*.%s.svc.cluster.local", namespace)
```

Comment:
I think there are some constants/variables for svc.cluster.local in the config.
If not I'd try to put it there. I guess this is fixed in k8s env, but if it can be different, better to have them in config.